crates/python/src/api.rs

```diff
@@ -102,6 +102,14 @@ py_wrap_error!(
     PyRuntimeError
 );
 
+/// Get the keyword `key` value from `kwds` if it is of type `Option<T>` and it is present, else `None`.
+/// Returns an error if a value is present but cannot be extracted into `T`.
+fn get_kwd<'a, T: FromPyObject<'a>>(kwds: Option<&'a PyDict>, key: &str) -> PyResult<Option<T>> {
+    kwds.and_then(|kwds| kwds.get_item(key))
+        .map(<Option<T>>::extract)
+        .unwrap_or(Ok(None))
```

Comment:
Can you get away with `unwrap_or_default()`?

Reply:
Unfortunately no - I think `Ok(None)` is probably the sanest default but makes sense that there isn't any since it's a bit arbitrary
```
`Result<std::option::Option<T>, pyo3::PyErr>: Default` is not satisfied
```